src/java.base/share/classes/java/lang/invoke/DirectMethodHandle.java

```diff
@@ -416,9 +416,13 @@ MethodHandle viewAsType(MethodType newType, boolean strict) {
         }
         Object checkReceiver(Object recv) {
             if (!caller.isInstance(recv)) {
-                String msg = String.format("Receiver class %s is not a subclass of caller class %s",
-                                           recv.getClass().getName(), caller.getName());
-                throw new IncompatibleClassChangeError(msg);
+                if (recv != null) {
+                    String msg = String.format("Receiver class %s is not a subclass of caller class %s",
+                                               recv.getClass().getName(), caller.getName());
+                    throw new IncompatibleClassChangeError(msg);
+                } else {
+                    throw new NullPointerException();
```

Comment:
Since we know which `member` this is trying to invoke, the **NPE** can be constructed with a [JEP 358](https://openjdk.java.net/jeps/358)‑style message.